### PR TITLE
feat: upgraded to node v18, added .nvmrc and updated workflows

### PR DESCRIPTION
**Pull request by BilalQamar95**
## Ticket
[Upgrade Node Js from 16 to 18](https://github.com/openedx/frontend-lib-special-exams/issues/91)

## Description
- Added support for node v18, updated .nvmrc and workflows. 
- Updated `frontend-platform` & `frontend-build` versions
- Resolved respective `eslint` issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +43,8 @@
     "url": "https://github.com/edx/frontend-lib-special-exams/issues"
   },
   "dependencies": {
+    "@edx/frontend-build": "12.7.0",
+    "@edx/frontend-platform": "4.0.1",
```
Comment (OmarIthawi) [inline package.json:47]:
```suggestion
    "@edx/frontend-platform": "4.1.0",
```

**Review by OmarIthawi — APPROVED** [review]
Thanks. Now there's a more up to date version.

File: package.json (hunk 1 above)
Comment (brian-smith-tcril) [reply package.json:47]:
```suggestion
    "@edx/frontend-platform": "4.2.0",
```
Looks like the latest version of platform is now `4.2.0`, just a note to make sure to bump to whatever the latest is when merge conflicts are resolved. https://github.com/openedx/frontend-platform/releases

**Review by arbrandes — APPROVED** [review]
Looks great, approved!

**Pull request merged**